File: Lab_12_Angular_P2_CRUD/news-portal-app/src/app/shared/models/article.ts
export interface Article {
  id: number;
  title: string;
  content: string;
  publishedAt: string;
  categoryId: number;
  categoryName: string;
  authorId: string | null;
  authorName: string;
  imagePath?: string | null;
  tags: Tag[];
}

export interface Category {
  id: number;
  name: string;
}

export interface Tag {
  id: number;
  name: string;
}

export interface CurrentUser {
  id: string;
  name: string;
  email: string;
  roles: string[];
}

export interface CreateArticleDto {
  title: string;
  content: string;
  categoryId: number;
  tagIds?: number[];
}

export interface UpdateArticleDto {
  title: string;
  content: string;
  categoryId: number;
  tagIds?: number[];
}

File: Lab_12_Angular_P2_CRUD/backend/Models/Article.cs
namespace Lab12.Models;

using System.ComponentModel.DataAnnotations;

public class Article : BaseEntity
{
    [Required]
    [MinLength(5)]
    public string Title { get; set; } = string.Empty;

    [Required]
    [MinLength(20)]
    public string Content { get; set; } = string.Empty;

    [DataType(DataType.Date)]
    public DateTime PublishedAt { get; set; } = DateTime.UtcNow;

    public int CategoryId { get; set; }
    public Category? Category { get; set; }

    // Identity — AuthorId este string (GUID), nu int
    public string? AuthorId { get; set; }
    public ApplicationUser? Author { get; set; }

    public string? ImagePath { get; set; }

    // Navigare Many-to-Many catre Tag
    public List<Tag> Tags { get; set; } = new();
}
